src/emitter.rs

```diff
@@ -53,13 +54,13 @@ pub trait Emitter {
     fn emit_string(&mut self, c: &str);
 
     /// Set the _current token_ to a start tag.
-    fn init_start_tag(&mut self);
+    fn init_start_tag(&mut self, reader: &R);
```

Comment:
Adding reader everywhere is extremely unfortunate. Let's say some implementor of emit_string wants a Reader at some point: it would be a breaking change to add it to the method signature

also this change by itself is a breaking change

not entirely sure what to do about it, need more time to decide

Reply:
>Adding reader everywhere is extremely unfortunate.

I found it a quite elegant solution because it keeps the `Emitter` trait unaware of spans.

>Let's say some implementor of emit_string wants a Reader at some point: it would be a breaking change to add it to the method signature

If you want to cater to unforeseen future use cases we could add the `reader: &R` parameter to all `Emitter` methods.

>also this change by itself is a breaking change

Yes I know. But the crate is still 0.2, so I don't think it should be that big of a deal? You said yourself that your API is still "very in-flux".

Reply:
yes just thinking out loud, no better solution in sight

Reply:
>If you want to cater to unforeseen future use cases we could add the reader: &R parameter to all Emitter methods.

I think that's probably good, yeah